### PR TITLE
[Sc-479] Bug(M-6): Asserter can post invalid assertions with minimal penalty

### PR DESCRIPTION
**Pull request by 0xNuggan**
## What has been done
- Added the possibility to manually set the required Bond when setting the currency (enforcing it to be higher or equal to the minimum). This way users can set a higher one if they believe the funds at stake merit it.
- Adapted corresponding init functions and tests

### CODE REVIEW

File: src/modules/logicModule/abstracts/oracleIntegrations/UMA_OptimisticOracleV3/OptimisticOracleIntegrator.sol [hunk 1]
```diff
@@ -96,11 +98,11 @@ abstract contract OptimisticOracleIntegrator is
     // Setter Functions
 
     /// @inheritdoc IOptimisticOracleIntegrator
-    function setDefaultCurrency(address _newCurrency)
+    function setDefaultCurrency(address _newCurrency, uint _newBond)
```
Comment (Zitzak) [inline src/modules/logicModule/abstracts/oracleIntegrations/UMA_OptimisticOracleV3/OptimisticOracleIntegrator.sol]: Not strictly necessary but could be changed to reflect additional usecase. `setDefaultCurrencyAndBond`

**Review by Zitzak — APPROVED** [review]
LGTM. The one comment is optional I would say

File: src/modules/logicModule/abstracts/oracleIntegrations/UMA_OptimisticOracleV3/OptimisticOracleIntegrator.sol (hunk 1 above)
Comment (0xNuggan) [reply src/modules/logicModule/abstracts/oracleIntegrations/UMA_OptimisticOracleV3/OptimisticOracleIntegrator.sol]: Fair enough, changed it

**Review by FHieser — APPROVED** [review]
The implementation seems valid, but the bond basically means min amount of Tokens that will be distributed correct?

**Comment by 0xNuggan** [comment]
No, it's an amount that gets taken from the asserter in case of dispute, and as fee

**Pull request merged**